pkg/resource/training_job/custom_delta.go

```diff
@@ -46,4 +46,34 @@ func customSetDefaults(
 			a.ko.Spec.OutputDataConfig.KMSKeyID = b.ko.Spec.OutputDataConfig.KMSKeyID
 		}
 	}
+
+	// Default value of VolumeSizeInGB is 0
+	if ackcompare.IsNotNil(a.ko.Spec.ProfilerRuleConfigurations) && ackcompare.IsNotNil(b.ko.Spec.ProfilerRuleConfigurations) {
+		for index := range a.ko.Spec.ProfilerRuleConfigurations {
+			if ackcompare.IsNil(a.ko.Spec.ProfilerRuleConfigurations[index].VolumeSizeInGB) && ackcompare.IsNotNil(b.ko.Spec.ProfilerRuleConfigurations[index].VolumeSizeInGB) {
```

Comment:
I'm not sure if this is completely correct. What if the arrays have different sizes?

Reply:
Also i don't think you'll ever see a `nil` element in an array.

Reply:
the nil check on line 53 is in reference to VolumeSizeInGB which is an element of a map. As its an array of maps/dicts.